packages/core/src/test/amazonqFeatureDev/session/session.test.ts

```diff
@@ -31,4 +40,77 @@ describe('session', () => {
             })
         })
     })
+    describe('insertChanges', async () => {
+        const controllerSetup: ControllerSetup = await createController()
+
+        const workspaceFolder = controllerSetup.workspaceFolder
+
+        beforeEach(async () => {
+            const uploadID = '789'
+            const tabID = '123'
+            const workspaceFolders = [controllerSetup.workspaceFolder] as CurrentWsFolders
+
+            sinon.stub(VirtualFileSystem.prototype)
+
+            const testConfig = {
+                conversationId: conversationID,
+                proxyClient: {
+                    createConversation: () => sinon.stub(),
+                    createUploadUrl: () => sinon.stub(),
+                    generatePlan: () => sinon.stub(),
+                    startCodeGeneration: () => sinon.stub(),
+                    getCodeGeneration: () => sinon.stub(),
+                    exportResultArchive: () => sinon.stub(),
+                } as unknown as FeatureDevClient,
+                sourceRoots: [''],
+                uploadId: uploadID,
+                workspaceFolders,
+            }
+            const testApproach = 'test-approach'
+
+            const codeGenState = new CodeGenState(
```

Comment:
I think in the future we should definitely start creating some easier ways to create codegen state (and states in general). So many defaults store between here and the actual session stuff that we could just extract away when they aren't being manually set